src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -343,6 +397,65 @@ public void onDestroyed()
 
         ChunkDataHelper.claimColonyChunks(colony, false, this.getID(), getClaimRadius(getBuildingLevel()));
         ConstructionTapeHelper.removeConstructionTape(getCorners(), world);
+
+        for (final IBuildingModule module : modules.values())
+        {
+            module.onDestroyed();
+        }
+    }
+
+    @Override
+    public void removeCitizen(final ICitizenData citizen)
+    {
+        for (final IBuildingModule module : modules.values())
+        {
+            module.removeCitizen(citizen);
+        }
+        super.removeCitizen(citizen);
+    }
+
+    @Override
+    public boolean assignCitizen(final ICitizenData citizen)
```

Comment:
How do you see modules working together here? Specifically, the restaurant has two citizens with very different jobs. I expect that will be two modules, each with one citizen assigned. Is it First one wins? or do we expect a priority, and thus support promotion? If the cook dies, does the assistant cook become the cook?

Reply:
For now, It would be by addition order (so first cook than assistant, if cook dies, assistant stays assistant and new worker will become cook). However, yes, there is room for additional functionality in the future here.